### PR TITLE
new section: filter function's implementation

### PR DESCRIPTION
**Pull request by cliffordfajardo**
this addition is simply for learners who would like to understand how the map function works behind the scenes. This is one of many ways to implement it. It's not magic, it's javascript =).

### CODE REVIEW

File: JS-Array-Prototype-Filter.md [hunk 1]
```diff
@@ -66,3 +66,27 @@ var result = arr.filter(
...
+}
+
+function filterBy(value){
+  if(typeof value === 'number'){
+    return true;
+  }
+}
+
+let numbersArray = filter([1,'a',2, 'b'], filterByNumber); // numbersArray now holds --> [1,2]
+```
```
Comment (atjonathan) [inline JS-Array-Prototype-Filter.md]: :rocket: Add a [repl](https://repl.it) snippet, **`:rocket: [Run Code](url)`**

File: JS-Array-Prototype-Filter.md [hunk 2]
```diff
@@ -66,3 +66,27 @@ var result = arr.filter(
 document.write(result);
 // Output: element, the
 ```
+
```
Comment (atjonathan) [inline JS-Array-Prototype-Filter.md]: Remove this line.

File: JS-Array-Prototype-Filter.md [hunk 3]
```diff
@@ -66,3 +66,27 @@ var result = arr.filter(
...
+
+## How to create your own filter function for learning purposes
```
Comment (abhisekp) [inline JS-Array-Prototype-Filter.md]: ## Creating custom filter function

----
```md
## Creating custom filter function
```

File: JS-Array-Prototype-Filter.md [hunk 4]
```diff
@@ -66,3 +66,27 @@ var result = arr.filter(
...
+      filteredResults.push(array[i]);
+    }
+  }
+  return filteredResults;
+}
+
+function filterBy(value){
+  if(typeof value === 'number'){
+    return true;
+  }
```
Comment (abhisekp) [inline JS-Array-Prototype-Filter.md]:
```js
function filterBy(value) {
  return typeof value === 'number'
}
```

File: JS-Array-Prototype-Filter.md [hunk 5]
```diff
@@ -66,3 +66,27 @@ var result = arr.filter(
...
+}
```
Comment (abhisekp) [inline JS-Array-Prototype-Filter.md]:
```js
function filterByNumber(value) {
  return typeof value === 'number'
}
```

File: JS-Array-Prototype-Filter.md [hunk 6]
```diff
@@ -66,3 +66,27 @@ var result = arr.filter(
...
+  let filteredResults = [];
+  for(let i = 0; i < array.length; i++){
+    //apply the passed in callback on each element in the array
+    //if the filter function passed, then push value to filteredResults array.
+    if(callback(array[i])){
+      filteredResults.push(array[i]);
+    }
+  }
+  return filteredResults;
+}
```
Comment (abhisekp) [inline JS-Array-Prototype-Filter.md]:
```js
function filter(array, predicateFn) {
  const filteredArray = [] 
  
  for (const i = 0; i < array.length; ++i) {
    const currEl = array[i] // current element
    if ( predicateFn(currEl) ) {
        filteredArray.push( currEl )
    }
  }

  return filteredArray
}
```

File: JS-Array-Prototype-Filter.md [hunk 7]
```diff
@@ -66,3 +66,27 @@ var result = arr.filter(
 document.write(result);
```
Comment (abhisekp) [inline JS-Array-Prototype-Filter.md]: Not concerning your PR but this should be `console.log`

**Comment by Rafase282** [comment]
@cliffordfajardo please create the articles at the [forum](http://forum.freecodecamp.com/) from now on.

**Pull request closed without merging**